test/e2e/common/node/pod_resize.go

```diff
@@ -71,1205 +71,543 @@ func offsetMemory(index int64, value string) string {
+				for i, c := range expectedContainers {
+					// For burstable pod test cases with initContainers, we only resize the initContainer.
+					if hasInitCtrs && !c.InitCtr {
+						continue
+					}
+					// For containers where the resize policy is "restart", we expect a restart.
+					expectRestart := int32(0)
+					if cpuPolicy == v1.RestartContainer && (desiredCPU != originalCPU || desiredCPULimit != originalCPULimit) {
+						expectRestart = 1
+					}
+					if memPolicy == v1.RestartContainer && (desiredMem != originalMem || desiredMemLimit != originalMemLimit) {
+						expectRestart = 1
+					}
+					c.RestartCount = expectRestart
+					expectedContainers[i] = c
+				}
+
+				ginkgo.By("patching and verifying pod for resize")
+				patchAndVerify(ctx, f, podClient, newPod, originalContainers, expectedContainers, "resize")
+
+				// Resize has been actuated, test the reverse operation.
+				rollbackContainers := createRollbackContainers(originalContainers, expectedContainers)
+				ginkgo.By("patching and verifying pod for rollback")
+				patchAndVerify(ctx, f, podClient, newPod, expectedContainers, rollbackContainers, "rollback")
+
+				ginkgo.By("deleting pod")
+				podClient.DeleteSync(ctx, newPod.Name, metav1.DeleteOptions{}, f.Timeouts.PodDelete)
+			},
+				// All tests will perform the requested resize, and once completed, will roll back the change.
+				// This results in the coverage of both increase and decrease of resources.
+				ginkgo.Entry("cpu req increase, then decrease", increasedCPU, originalCPULimit, originalMem, originalMemLimit),
+				ginkgo.Entry("cpu limit increase, then decrease", originalCPU, increasedCPULimit, originalMem, originalMemLimit),
+				ginkgo.Entry("mem req increase, then decrease", originalCPU, originalCPULimit, increasedMem, originalMemLimit),
+				ginkgo.Entry("mem limit increase, then decrease", originalCPU, originalCPULimit, originalMem, increasedMemLimit),
+				ginkgo.Entry("increase all, then decrease", increasedCPU, increasedCPULimit, increasedMem, increasedMemLimit),
+				ginkgo.Entry("increase cpu & decrease mem, then decrease cpu & increase mem", increasedCPU, increasedCPULimit, reducedMem, originalMemLimit),
+			)
+		},
+			ginkgo.Entry("no restart", v1.NotRequired, v1.NotRequired, false),
+			ginkgo.Entry("no restart, resize initContainer only", v1.NotRequired, v1.NotRequired, true),
+			ginkgo.Entry("cpu restart", v1.RestartContainer, v1.NotRequired, false),
+			ginkgo.Entry("mem restart", v1.NotRequired, v1.RestartContainer, false),
+			ginkgo.Entry("both cpu and mem restart", v1.RestartContainer, v1.RestartContainer, false),
+			ginkgo.Entry("both cpu and mem restart, resize initContainer only", v1.RestartContainer, v1.RestartContainer, true),
+		)
+	})
+
+	ginkgo.DescribeTable("burstable pods - extended", func(ctx context.Context, originalContainers, expectedContainers []podresize.ResizableContainerInfo, doRollback bool) {
```

Comment:
nit: can you give this a more descriptive name, or add a comment

Reply:
added a comment